### PR TITLE
Fixes #1002 Error when double-clicking on a molecule

### LINKED ISSUE #1002: Error when double-clicking on a molecule
Expand Modules -> <Module> -> Molecules BB -> double-click on any molecule

image

Application:
MoBi® 11.2.471

Object reference not set to an instance of an object.

Stack trace:
```
at MoBi.Presentation.Presenter.EditMoleculeBuilderPresenter.Edit(MoleculeBuilder moleculeBuilder)
   at MoBi.Presentation.Presenter.EditMoleculeBuildingBlockPresenter.editPresenter[T](T objectBase, IEditPresenterWithParameters`1 editPresenter, IParameter parameter)
   at MoBi.Presentation.Presenter.EditMoleculeBuildingBlockPresenter.setupEditPresenterFor(IObjectBase objectBase, IParameter parameter)
   at MoBi.Presentation.Presenter.EditMoleculeBuildingBlockPresenter.SelectBuilder(MoleculeBuilder builder)
   at MoBi.Presentation.Presenter.EditBuildingBlockPresenterBase`4.selectObjectAndParent(IContainer parentObject, IObjectBase selectedObject)
   at MoBi.Presentation.Presenter.EditBuildingBlockPresenterBase`4.Handle(EntitySelectedEvent eventToHandle)
   at OSPSuite.Utility.Events.EventPublisher.<>c__DisplayClass4_1`1.<PublishEvent>b__1()
   at OSPSuite.Utility.Exceptions.ExceptionManagerBase.Execute(Action action)
```

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

File: src/MoBi.Presentation/Presenter/EditMoleculeBuilderPresenter.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public void Edit(MoleculeBuilder moleculeBuilder, IReadOnlyList<IObjectBase> exi
 
       public void Edit(MoleculeBuilder moleculeBuilder)
       {
-         Edit(moleculeBuilder, moleculeBuilder.ParentContainer.Children);
+         Edit(moleculeBuilder, moleculeBuilder.ParentContainer?.Children);
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/Presenter/EditMoleculeBuilderPresenter.cs:108]: It was just this ParentContainer can be null. In develop branch, a null value is acceptable in the second parameter of Edit

Comment (msevestre) [reply src/MoBi.Presentation/Presenter/EditMoleculeBuilderPresenter.cs:108]: what's the parent container of a molecule builder? I don't get it

**Comment by msevestre** [comment]
I don't understand this fix

**Comment by rwmcintosh** [comment]
The fix is basically just a null check. Previously code called Edit with a null second parameter. I am assuming that there is such a thing as ParentContainer for MoleculeBuilder in some circumstances but I have no idea about that.

https://github.com/Open-Systems-Pharmacology/MoBi/commit/4360c402bc2cdc5725c82cdfb22dfd627545409e#diff-abbed452b6199955351c2fa19858ec9f125d3723c2f1d9e343787bfa7619d7c2L113-L116

**Comment by msevestre** [comment]
What I don't understand is why OSMOSES is different. What has changed?

**Comment by rwmcintosh** [comment]
The link there is to some lines that were changed from refactoring ObjectBaseDTO commit.

image

It's far down in the diff so you have to wait for a second until the lines load and are highlighted.

Previously (and in develop branch) the ```.Children``` was not taken.

**Pull request merged**